test/integration/streamCounter.test.js

```diff
@@ -0,0 +1,51 @@
+import got from 'got';
+import random from 'random';
+
+const testUserId = 'test_user_1987';
+const randomTestUserId = random.int(0, 1000000);
+
+describe('Test streamService endpoints', () => {
+    afterEach(async () => {
+        got.post(`http://localhost:3000/stream/removeReservation/${testUserId}`);
+        got.post(`http://localhost:3000/stream/removeReservation/${testUserId}`);
+        got.post(`http://localhost:3000/stream/removeReservation/${testUserId}`);
+    });
+
+    it('First time request for a stream should return 200 response', async () => {
+        const response = await got.post(`http://localhost:3000/stream/reservation/${testUserId}`);
+        expect(response.statusCode).toBe(200);
+    });
+
+    it('Exceeding the number of streams should result in 400 being returned', async () => {
+        await got.post(`http://localhost:3000/stream/reservation/${testUserId}`);
+        await got.post(`http://localhost:3000/stream/reservation/${testUserId}`);
+        await got.post(`http://localhost:3000/stream/reservation/${testUserId}`);
+        let statusCode;
+        try {
+            await got.post(`http://localhost:3000/stream/reservation/${testUserId}`);
+        } catch(e) {
+            statusCode = e.response.statusCode;
+        }
+        expect(statusCode).toBe(400);
+    });
+
+    it('Requesting stream 3 times and then one request to remove one stream should result in 200 being returned', async () => {
```

Comment:
```suggestion
    it('Requesting stream 3 times, removing one reservation and then requesting the reservation again should result in 200 being returned', async () => {
```
I think testing the behaviour like that makes more sense - it not only tests if the second endpoint returns 200 response, but also if it actually alters the value in the db as expected

Reply:
Yes and to be fair this is what you originally suggested for this test to be in the first place anyway :)